source/initial-dns-seedlist-discovery.rst

```diff
@@ -0,0 +1,247 @@
+
+1. Parsed protocol: ``mongodb+srv``
+   
+   Parsed hostname: ``test1.test.mongodb.com``
+
+2. The following servers MUST now be known to SDAM:
+   
+   ``localhost:27017`` ``localhost:27018`` ``localhost:27019``
+
+3. The driver should be aware that the following seed hosts exist:
+   
+   ``localhost:27017`` ``localhost:27018``
+
+Two results with non-standard port
+----------------------------------
+
+``mongodb+srv://test2.test.mongodb.com/``
+
+1. Parsed protocol: ``mongodb+srv``
+
+   Parsed hostname: ``test2.test.mongodb.com``
+
+2. The following server/port combinations must now be known to SDAM:
+
+   ``localhost:27017`` ``localhost:27018`` ``localhost:27019``
+
+3. The driver should be aware that the following seed hosts exist:
+
+   ``localhost:27018`` ``localhost:27019``
+
+One result with default port
+----------------------------
+
+``mongodb+srv://test3.test.mongodb.com/``
+
+1. Parsed protocol: ``mongodb+srv``
+
+   Parsed hostname: ``test3.test.mongodb.com``
+
+2. The following server/port combinations must now be known to SDAM:
+
+   ``localhost:27017 localhost:27018 localhost:27019``
+
+3. The driver should be aware the following seed host exist:
+
+   ``localhost:27017``
+
+No results
+----------
+
+``mongodb+srv://test4.test.mongodb.com/``
+
+
+1. Parsed protocol: ``mongodb+srv``
+
+   Parsed hostname: ``test4.test.mongodb.com``
+
+2. No error/exception raised constructing a MongoClient object
+
+3. An error/exception is raised when doing the ping operation, with the
+   message that there were no SRV records found for ``test4.test.mongodb.com``.
+
+Multiple Names in URI
+---------------------
+
+``mongodb+srv://test5.test.mongodb.com,test6.test.mongodb.com/``
+
+1. Parser must fail while constructing the MongoClient object, because two
+   hostnames are specified with the ``mongodb+srv://`` protocol
+
+2. The driver MUST NOT run the operation and MUST NOT be aware of any hosts
+
+URI has a port
+--------------
+
+``mongodb+srv://test7.test.mongodb.com:27018/``
+
+1. Parser must fail while constructing the MongoClient object, because a port
+   is specified with the ``mongodb+srv://`` protocol
+
+2. The driver MUST NOT run the operation and MUST NOT be aware of any hosts
+
+Motivation
+==========
+
+Several of our users have asked for this through tickets:
+
+* `<https://jira.mongodb.org/browse/DRIVERS-201>`_
+* `<https://jira.mongodb.org/browse/NODE-865>`_
+* `<https://jira.mongodb.org/browse/CSHARP-536>`_
+
+Design Rationale
+================
+
+The design specifically calls for a pre-processing stage of the processing of
+connection URLs to minimize the impact on existing functionality.
+
+Reference Implementation
+========================
+
+None yet.
+
+Backwards Compatibility
+=======================
+
+There are no backwards compatibility concerns.
+
+Future Work
+===========
+
+In the future we could consider using the priority and weight fields of the
+SRV records, or to use SRV records to do mongos discovery.
```

Comment:
"MongoS" is used in the abstract. Should these be consistent?

Reply:
I prefer it, `mongos` is very hard for me to skim visually